otter/convergence/transforming.py

```diff
@@ -0,0 +1,117 @@
+"""
+Functions for transforming collections of steps.
+
+There are a few classes of transformations here:
+
+- optimizing (converting multiple steps into one for the purposes of reducing
+  API roundtrips)
+- limiting (by truncating the number of steps we take)
+- throttling (by wrapping multiple steps with one that gradually executes them)
+"""
```

Comment:
E.g. I'm not disagreeing with that perspective, just pointing out another way to look at it as well.  :)  I don't feel strongly whichever way we go.